### DISCUSSION TITLE
Let's Take Revolt To The Next Level!  We Can Do This! :)

### DISCUSSION THREAD

**Discussion by TrazhCant** [stoatchat/discussions#881] [Feedback]
I can't take credit for the entirety of the content in this post save for the time I spent posing the questions and sharing them here.  I had a chat with Grok AI about strategies for Revolt's growth, specifically on what kind of donation levels might make a meaningful impact for Revolt (i.e. give it a little more gas and punch!).  Then followed up with discussing relevant strategies for accomplishing exactly that.  I hope this helps.

I'm not a dev and I'm poor, but I also know a lot about tech and of the demographics that are exploited and preyed upon the most year over year (most of us think it's granny getting scam calls from overseas) it's actually gamers.  Gamers need Revolt to rival the options they currently (and obliviously, I'd argue) leverage to try and enjoy doing what they do with friends/family/etc.  I don't infer that these closed-source, for-profit platforms need to be "run out of business" or anything.  The problem is (and I'd argue always is in just about anything) choice or the lack thereof.  The vulnerable need to have the ability to choose something that's meaningfully different at a fundamental level.  (In this case, open-source vs. closed-source).

That concludes my preface.  Much love to all.  I hope this is useful, or at the very least intriguing. ;)

> ### PROMPT:
> Open Source projects as you have mentioned rely on donations and volunteerism from devs many times.  Though it would vary from project to project, what kind of donation levels might bring a meaningful increase in fulfilling Revolt's roadmap goals?  From what I understand, there are 3 main developers on the project currently.
> 
> ### GROK ANSWER:
> Determining exact donation levels that would meaningfully advance Revolt's roadmap goals involves considering various factors such as the cost of development time, operational costs, infrastructure, and the specific goals on the roadmap. Here's a broad perspective based on common practices:
> 
> #### Developer Compensation: 
> 
> - If we consider the three main developers working full-time or at least part-time on Revolt, a rough estimate for a software developer's compensation can range widely. For a ballpark figure in the open-source context, where developers might work for significantly less than industry standards:
> 
>     - Part-Time: Assuming each developer works 20 hours a week at $30/hour (which is a modest rate for software developers, especially considering they might volunteer some time), that's $600 per week per developer, or $1,800 for all three. Annually, this would be around $93,600.
>     - Full-Time: At 40 hours a week, this would double to $3,600 weekly or $187,200 annually for all three.
> 
> #### Operational Costs: 
> 
> - Server and Infrastructure: Costs for hosting, bandwidth, and scaling up server capacity to support more users or features like screen sharing could easily run into thousands of dollars per month, especially if aiming for global reach and reliability.
> - Software Tools: Licenses for development tools, CI/CD services, etc.
> 
> #### Project Specific Goals:
> 
> - Implementing features like screen sharing would require investment in both time (for development) and infrastructure (for processing and bandwidth).
> - Feature Development: If screen sharing were a priority, this could be a significant line item. Let's say it requires 3 months of focused development by one developer, that's roughly $36,000 if we use the part-time compensation model.
> 
> #### Given these considerations:
> 
> - Monthly Donations: 
>   - To sustain three part-time developers, cover basic operational costs, and fund significant features like screen sharing, you might look at a starting point of:
>     - Low Impact: $5,000 - $10,000 per month could cover basic costs and part-time salaries for the developers.
>     - Moderate Impact: $15,000 - $25,000 per month would allow for more aggressive roadmap fulfillment, potentially including hiring additional help or scaling infrastructure.
>     - High Impact: $30,000+ per month would provide ample resources to significantly speed up development, enhance infrastructure, and possibly transition to full-time work for developers or expand the team.
> 
> - One-Time Donations: Large one-time donations can fund specific features or projects. For instance:
>   - A $50,000 donation might fund a significant feature like screen sharing if used specifically for that purpose, assuming efficient project management and existing infrastructure support.
> 
> These figures are speculative and can vary greatly based on project specifics, the cost of living in the developers' location, and the exact features prioritized. Revolt, like many open-source projects, would benefit from both recurring donations for sustainability and large donations for pushing forward specific, high-impact features. Remember, transparency and community involvement in how these funds are used also play a significant role in the success of such initiatives.
> 
> ---------------------------------------------------------------------------------------
> 
> ### PROMPT:
> In my opinion, the biggest problem with generating donations is going to be public awareness, and then of course the project needs to resonate meaningfully enough with said public that they choose to donate.  Secondarily, in my opinion, providing a mechanism for donors to OPT IN to an automatic/recurring donation would also be key.  Speaking on Revolt specifically, how would you suggest the Revolt team accomplish these things?  (Raise awareness to appropriate audiences, and the best way to adapt their current means of accepting donations to support an OPT IN automatic/recurring donation from donors).  Be as detailed as possible.
> 
> ### GROK ANSWER:
> Here are detailed strategies for Revolt to raise awareness and implement an opt-in automatic/recurring donation system:
> 
> #### Raising Awareness:
> - Targeted Marketing to Gamers and Tech Communities:
>   - Content Creation: Produce content like tutorials, feature breakdowns, or comparisons with Discord on platforms like YouTube, Twitch, and gaming blogs. Engage with gaming influencers or streamers who might appreciate an open-source alternative.
>   - Gaming Forums and Social Media: Actively participate in forums like Reddit's r/gamedev, r/pcgaming, or Discord communities. Share updates, participate in discussions, and highlight Revolt's unique features or privacy focus.
>   - Open Source Events: Attend or sponsor events like hackathons, tech meetups, or open-source conferences where you can demonstrate Revolt. Offer booths, talks, or workshops.
> - Leverage Open Source Ecosystem:
>   - Collaborations: Team up with other open-source projects or tools that gamers use. Cross-promotion through newsletters, blogs, or joint streams can be effective.
>   - Open Source Platforms: Utilize GitHub, GitLab, or similar platforms not just for development but also for marketing. Keep the project page updated with news, badges for contributors, and calls for support.
> - Content Marketing:
>   - Blogging: Write articles on the Revolt blog about the project's progress, the philosophy behind it, and the benefits of open-source communication for gamers. SEO optimization can help reach those searching for Discord alternatives.
>   - Case Studies: Share stories of how Revolt is used by gaming communities or developers, highlighting its advantages.
> - Community Engagement:
>   - Discord Presence: Ironically, maintain a presence on Discord to reach those already within the ecosystem, explaining how Revolt offers similar experiences with added benefits like privacy or control.
>   - Feedback Channels: Use Revolt itself for community feedback, making users feel part of the development process, which can foster loyalty and advocacy.
> - Social Proof:
>   - Testimonials: Collect and share testimonials from users or notable personalities in the gaming or tech world who support Revolt.
> Showcase Features: Regularly update the community with new features or improvements, making sure each update is widely announced.
> 
> #### Implementing Automatic/Recurring Donations:
> - Choose a Donation Platform:
>   - Open Collective: Already used by many open-source projects, it supports recurring donations and provides a transparent way to manage funds. It's community-focused, which resonates well with open-source ethos.
>   - Patreon for Open Source: Although more commonly associated with creatives, Patreon can be adapted for software projects, offering tiers with different rewards or recognition levels.
>   - GitHub Sponsors: If not already integrated, this could leverage the existing developer base on GitHub, offering an easy way for developers to support Revolt.
> - Implementation Steps:
>   - Integration: Integrate the chosen platform's API or widget into the Revolt website, making it easy for users to set up recurring donations. Ensure this process is as simple as possible, ideally one or two clicks from the main page.
>   - Tiers and Benefits: Establish donation tiers with clear benefits. Even if benefits are symbolic (like special roles in the community server, early access to features, or name in credits), they can motivate recurring donations.
>   - Transparency: Regularly update donors on how funds are used, perhaps through a dedicated section on the website or updates in the community server. Transparency can increase trust and encourage ongoing support.
> - Promote the Donation System:
>   - Call to Action: Have clear, prominent calls to action on the website, within the app, and in all communications. Use compelling language like "Support the future of open communication."
>   - Reminders: Occasionally remind users of the benefits of their support through announcements or during significant updates or milestones.
>   - Donor Recognition: Publicly thank donors or highlight their contributions in community channels or on social media (with their permission).
> - User Experience:
>   - Ease of Use: Make sure the donation process is seamless across devices. Users should be able to set up recurring donations without much friction.
>   - Privacy and Security: Ensure that all donation transactions are secure and that users feel their data is protected.
> 
> By combining these strategies, Revolt can increase both its visibility and the sustainability of its funding model, fostering a community that not only uses but also supports the project financially.
> 
> ---------------------------------------------------------------------------------------
> 
> ### PROMPT:
> Does Revolt's current means of accepting donations already support recurring automatic donations?
> 
> ### GROK ANSWER:
> As of the latest information available, Revolt's current means of accepting donations do not explicitly support recurring automatic donations. They primarily accept one-time donations through platforms like Open Collective. However, Open Collective does support recurring donations, so there's potential for Revolt to leverage this feature if they haven't already:
> 
> - Open Collective: This platform allows for setting up recurring donations. If Revolt has integrated Open Collective as their donation platform, they could enable this feature by setting up donation tiers that include options for monthly contributions. Users would then be able to opt into these recurring donations directly through the Open Collective page for Revolt.
> 
> If Revolt hasn't yet enabled this feature on Open Collective, here's what they would need to do:
> 
> - Enable Recurring Donations on Open Collective: Go into their Open Collective admin settings to turn on the recurring donation option for each tier or create new tiers specifically for recurring donations.
> - Promote Recurring Donations: Update their website, documentation, and social media to inform users about the option for recurring donations, explaining how it supports the ongoing development of Revolt.
> - Documentation and Integration: Make sure there's clear documentation or integration within Revolt's ecosystem (like on their website or within the app) that links to or explains how to set up recurring donations on Open Collective.
> 
> Since the information might change, and Revolt could implement this feature post my knowledge update, it's recommended to check directly on Revolt's official channels or their Open Collective page for the most current donation options.

**Comment by TrazhCant** [comment]
One additional idea that might be adaptable in some way is the recent approach KDE Plasma took by utilizing a notification popup that occurs only 1 time per year regarding donating to the project.  While I don't suggest Revolt use a "popup" specifically, what I'm pointing out is the built-in (integration) of a donation "thingie" which REQUIRES some level of user interaction even if it's to dismiss the notification.  This announcement from the KDE Plasma team was met with mixed opinions (quite emphatic and passionate btw) and turned out to be a HUGE success for KDE Plasma.  This  illustrates what I think is an acceptable, reverent and respectful approach to mandate the user's direct engagement on the topic of donating (even if it's to click "dismiss") without being a "Sheldon knocking on your door and calling your name incessantly until you answer".  By limiting the "Sheldon" shenanigans to only once per year, that doesn't compromise user-base retention, and is brilliantly effective if KDE Plasma's results are anything to go by.  :)

**Comment by AlbatrosHD** [comment]
Those are good ideas, thank you for the Post.
I hope there will be a way to donate to this Project one day. 
Even tho it's sometimes possible to donate to the Maintainers separately, which takes a bit to much effort for most people.

Hypothetically if every user (500.000 ish for simplicity) donates 1 buck, we would get 500k. Which won't be achieved but it's still a significant sum.  So please don't be shy to ask for donations, you are not alone dear Maintainers!

**Reply by AlbatrosHD** [reply]
And here is a way to donate: https://wiki.revolt.chat/notes/project/financial-support/

**Reply by TrazhCant** [reply]
Dropped a $20-spot to kick things off.  PLS CLICKS TEH LINKS PEEPS!  Droppa dalla droppa dime - every bit helps and can add up quick!  :)
